src/Symfony/Component/Finder/Finder.php

```diff
@@ -107,17 +108,21 @@ public function files()
      *
      *   $finder->depth('> 1') // the Finder will start matching at level 1.
      *   $finder->depth('< 3') // the Finder will descend at most 3 levels of directories below the starting point.
+     *   $finder->depth(['>= 1', '< 3'])
      *
-     * @param string|int $level The depth level expression
+     * @param string|int|string[]|int[] $levels The depth level expression or an array of depth levels
      *
      * @return $this
      *
      * @see DepthRangeFilterIterator
      * @see NumberComparator
      */
-    public function depth($level)
+    public function depth($levels)
     {
-        $this->depths[] = new Comparator\NumberComparator($level);
+        $this->depths = \array_merge(
+            $this->depths,
+            ComparatorFactory::createBatch(NumberComparator::class, (array) $levels)
```

Comment:
IMHO you can inline the `array_map` call per case, no need for `ComparatorFactory`

Reply:
You are right, the Factory did not please me either (it was on my PR TODO list).
I opt for the simplicity of the foreach loop.